static/js/helper-functions.js

```diff
@@ -15,4 +16,54 @@ function highlightClippings(wordToHighlight) {
 
         clipping.innerHTML = highlightedClipping;
     });
+}
+
+// ------- books.html -------
+// Hides books which doesn't match the search bar value
+function searchBook() {
+    const searchInput = document.querySelector('#search-book');
+    const books = document.querySelectorAll('.book-element');
+
+    function displayMatches() {
+        const regex = new RegExp(this.value, 'gi');
+
+        books.forEach(book => {
+            const title = book.querySelector('.book-title').textContent;
+            const author = book.querySelector('.book-author').textContent;
+
+            if (title.match(regex) || author.match(regex)) {
+                book.style.display = 'block';
+            }
+            else {
+                book.style.display = 'none';
+            }
+        })
+    }
+
+    searchInput.addEventListener('keyup', displayMatches);
+}
+
+// Switches between the list and the gallery book view + controls buttons' colors
+function switchView() {
+    const galleryBtn = document.querySelector('#btn-gallery');
+    const listBtn = document.querySelector('#btn-list');
+    const bookView = document.querySelector('#book-view');
+
+    function viewControl() {
+        buttonToDisable = (this === galleryBtn) ? listBtn : galleryBtn;
+
+        // Swap buttons' colors
+        buttonToDisable.classList.remove('btn-dark');
+        buttonToDisable.classList.add('btn-light');
+
+        this.classList.remove('btn-light');
+        this.classList.add('btn-dark');
+
+        // Switch the book view
+        bookView.className = (this === galleryBtn) ? 'gallery-view' : 'list-view';
+    }
+
+    galleryBtn.addEventListener('click', viewControl);
+    listBtn.addEventListener('click', viewControl);
```

Comment:
Random notes (I don't think we have to do this, just want to share my braindump): It'd be nicer to have only one listener on a parent element and then distinguish the button based on the click target, wouldn't it? 

The class swapping could also be simplified to just deleting the "active" class from all child elements and then add it again to the element of the click target. 

The class name of the view ("`gallery-view`" and "`list-view`") could also be data attributes at the button HTML elements. Then the view control JavaScript would be 100% generic and re-usable, and adding a 3rd view is as easy as adding another button with an arbitrary class name in the data attributes 😊

Reply:
> It'd be nicer to have only one listener on a parent element and then distinguish the button based on the click target, wouldn't it?

The problem here is that the buttons have 2 nested elements, so to make sure that I've got a button I would have to check every time what've got a then get the parent or the parent's parent element if needed. It seems a little bit overkill and I'm afraid that it would make the code more messy.

---

> The class swapping could also be simplified to just deleting the "active" class from all child elements and then add it again to the element of the click target.

btn-light and btn-dark are bootstrap's classes so the only way to implement active funcitonality would be to copy css styles of those classes. So there are 3 solutions here:
1. The current one,
2. Copy bootstrap's classes,
3. This one (easier to extend):
    ```javascript
    Array.from(this.parentElement.children).forEach(btn => {
        btn.classList.remove('btn-dark');
        btn.classList.add('btn-light');
    })
    
    this.classList.remove('btn-light');
    this.classList.add('btn-dark');
    ```

---

> could also be data attributes at the button HTML elements

You are right, this is much cleaner solution!

Reply:
I think I need your opinion only on this one

Reply:
> btn-light and btn-dark are bootstrap's classes so the only way to implement active funcitonality would be to copy css styles of those classe

True. The "problem" is basically, that we actually have a binary state (active/inactive), but no binary CSS class that is either added or not.

Let's keep it simple for now "and ship it", we might have some follow-up updates and potential UI tweaks anyways.